### DISCUSSION TITLE
Error communicating with OpenAI | API key checked

### DISCUSSION THREAD

**Discussion by votkon** [ajndkr/lanarky#44] [Q&A] [answered]
Hi,
I am having trying to run the [conversation_chain.py](https://github.com/ajndkr/lanarky/blob/main/examples/app/conversation_chain.py) example, but always got the following error:

Retrying langchain.chat_models.openai.acompletion_with_retry.<locals>._completion_with_retry in 1.0 seconds as it raised APIConnectionError: Error communicating with OpenAI.
Retrying langchain.chat_models.openai.acompletion_with_retry.<locals>._completion_with_retry in 2.0 seconds as it raised APIConnectionError: Error communicating with OpenAI.
Retrying langchain.chat_models.openai.acompletion_with_retry.<locals>._completion_with_retry in 4.0 seconds as it raised APIConnectionError: Error communicating with OpenAI.
Retrying langchain.chat_models.openai.acompletion_with_retry.<locals>._completion_with_retry in 8.0 seconds as it raised APIConnectionError: Error communicating with OpenAI.
Retrying langchain.chat_models.openai.acompletion_with_retry.<locals>._completion_with_retry in 16.0 seconds as it raised APIConnectionError: Error communicating with OpenAI.
Error communicating with OpenAI

I've checked the API key and the connection - it all works correctly.
The conversation chain works ok too.

I can't figure out how I can this message more verbose to check it further.
I'm kinda lost at this point where should I investigate this further?

**Comment by ajndkr** [comment, marked as answer]
Hi! This retrying message shows up because of API rate limiting from OpenAI side. Are you using a free account?

This happens more frequently for free accounts. It can also be due some degraded performance of the API which causes more frequent connection drops. 

I wouldn’t worry too much about this. The retry mechanism handles this scenario.

**Reply by votkon** [reply]
Thanks for the quick reply! 
That's a paid account and the issue persists all the time :(
I've had these timeouts with ChatGPT in other use cases but was able to make it work by setting the timeout option.
Here I've tried to set the request_timeout to=120 for the chain, but it doesn't seem to make any difference.
It looks like this timeout logic is integrated somewhere inside of langchain module but I can't figure out how to change it there.

**Reply by votkon** [reply]
Maybe there's a way to make this error more verbose(to at least be sure that it times out) or override the timeout setting somehow?

**Reply by ajndkr** [reply]
I see! in that case, I recommend heading over to the langchain repo discussion. someone there might already have an answer ready. https://github.com/hwchase17/langchain/discussions

**Comment by votkon** [comment]
I found the solution for this: Python root TLS certs were not set up.
To fix do this:
pip3.10 install certifi
/Applications/Python\ 3.10/Install\ Certificates.command
